File: tests/unit/test_code_analysis/fixtures/python/clean.py
"""A well-structured module for testing quality metrics."""

from __future__ import annotations

from dataclasses import dataclass


TIMEOUT: int = 30


@dataclass
class User:
    """Represents an authenticated user."""

    name: str
    email: str
    active: bool = True

    def display_name(self) -> str:
        """Return formatted display name."""
        return f"{self.name} <{self.email}>"

    def deactivate(self) -> None:
        """Mark user as inactive."""
        self.active = False


def validate_email(email: str) -> bool:
    """Check if email has a valid format."""
    return "@" in email and "." in email.split("@")[1]


def create_user(name: str, email: str) -> User:
    """Factory function for User creation with validation."""
    if not validate_email(email):
        raise ValueError(f"Invalid email: {email}")
    return User(name=name, email=email)

File: tests/unit/test_code_analysis/fixtures/csharp/Clean.cs
using System;
using System.Collections.Generic;

namespace CleanApp
{
    /// <summary>Represents an authenticated user.</summary>
    public sealed class User
    {
        public string Name { get; }
        public string Email { get; }

        public User(string name, string email)
        {
            Name = name;
            Email = email;
        }

        /// <summary>Returns the formatted display name.</summary>
        public string DisplayName()
        {
            return $"{Name} <{Email}>";
        }

        /// <summary>Checks whether an email has a valid format.</summary>
        public bool HasValidEmail(string email)
        {
            if (string.IsNullOrEmpty(email))
            {
                return false;
            }
            return email.Contains("@");
        }
    }
}
